src/http-client.js

```diff
@@ -111,21 +112,26 @@ const checkParams = (name, payload, requires = []) => {
  * @param {object} headers
  * @returns {object} The api response
  */
-const publicCall = ({ endpoints }) => (path, data, method = 'GET', headers = {}) =>
-  sendResult(
+const publicCall = ({ proxy, endpoints }) => (path, data, method = 'GET', headers = {}) => {
+  const opts = {
+    method,
+    json: true,
+    headers,
```

Comment:
can you add the agent like so in this PR:

```
headers,
...(proxy ? { agent: new HttpsProxyAgent(proxy) } : {}),
```

Reply:
Oh I thought it was cleaner but yes ok